ports/glutin/window.rs

```diff
@@ -1040,74 +1064,64 @@ impl WindowMethods for Window {
 
     fn set_inner_size(&self, _: BrowserId, size: Size2D<u32>) {
         match self.kind {
-            WindowKind::Window(ref window) => {
-                window.set_inner_size(size.width as u32, size.height as u32)
-            }
-            WindowKind::Headless(..) => {}
+            WindowKind::Window(ref window, ..) => window.set_inner_size(size.width as u32, size.height as u32),
+            WindowKind::Headless(..) => {},
         }
     }
 
     fn set_position(&self, _: BrowserId, point: Point2D<i32>) {
         match self.kind {
-            WindowKind::Window(ref window) => {
-                window.set_position(point.x, point.y)
-            }
-            WindowKind::Headless(..) => {}
+            WindowKind::Window(ref window, ..) => window.set_position(point.x, point.y),
+            WindowKind::Headless(..) => {},
         }
     }
 
     fn set_fullscreen_state(&self, _: BrowserId, _state: bool) {
         match self.kind {
-            WindowKind::Window(..) => {
-                warn!("Fullscreen is not implemented!")
-            },
-            WindowKind::Headless(..) => {}
+            WindowKind::Window(..) => warn!("Fullscreen is not implemented!"),
+            WindowKind::Headless(..) => {},
         }
     }
 
     fn present(&self) {
         match self.kind {
-            WindowKind::Window(ref window) => {
+            WindowKind::Window(ref window, ..) => {
                 if let Err(err) = window.swap_buffers() {
                     warn!("Failed to swap window buffers ({}).", err);
                 }
-            }
-            WindowKind::Headless(..) => {}
+            },
+            WindowKind::Headless(..) => {},
         }
     }
 
     fn create_event_loop_waker(&self) -> Box<EventLoopWaker> {
         struct GlutinEventLoopWaker {
-            window_proxy: Option<glutin::WindowProxy>,
+            events_loop: Option<Arc<Mutex<glutin::EventsLoop>>>,
         }
         impl EventLoopWaker for GlutinEventLoopWaker {
             fn wake(&self) {
                 // kick the OS event loop awake.
-                if let Some(ref window_proxy) = self.window_proxy {
-                    window_proxy.wakeup_event_loop()
+                if let Some(ref events_loop) = self.events_loop {
+                    events_loop.lock().unwrap().create_proxy().wakeup();
```

Comment:
This will be easier if you use EventsLoopProxy. No need to create a proxy everytime.